### ISSUE TITLE
Feature Question: Couple Playcount with BlissMixer Results.

### ISSUE THREAD

**Issue by chincheta0815** [CDrummond/lms-blissmixer#3]
Hej,

is there a feature that selects a track based on the bliss results plus a playcount? This would be an option to make the tracks being based of ones ""favorites" or often played songs making the resulting playlist be more personal and "liked".

**Comment by CDrummond** [comment]
Not currently, but I have thought about having the plugin (this repo) ask for 75ish tracks from the mixer. The plugin would then read the playcounts for those tracks, sort based upon that, trim to the top (e.g.) 10 tracks, and then choose 5 random tracks from that.

**Comment by CDrummond** [comment]
Actually this might be possible if the mixer binary reads the lms db.

**Comment by chrober** [comment]
Hi @CDrummond,

this issue looked like a good place to point you at some related experimentation I have been doing around Bliss Mixer:

https://github.com/chrober/lms-blissmixer-lab

The repo is intentionally not a fork/replacement of Bliss Mixer. It is a sidecar plugin, **Bliss Mixer Lab**, that installs alongside the upstream plugin, reads the existing Bliss Mixer analysis database/preferences, and provides experimental features in a separate **Bliss (Lab)** DSTM provider.

One of the implemented experiments is directly related to this issue: play-count influence for DSTM selection. It is exposed as a slider that can go both ways:

- negative values favor less-played tracks
- positive values favor frequently played tracks
- zero disables the play-count influence

The selection still keeps Bliss/acoustic similarity as the primary basis; play count is an additional ranking influence rather than a replacement.

The lab plugin also currently experiments with learned-matrix weighting and Last.fm recording-similarity guidance. My goal is to use it as a staging area for early testing before proposing selected features for upstream Bliss Mixer, if they turn out to be useful and fit the direction of the original plugin.

I thought this might be useful context for this feature request, and perhaps a practical reference point if you decide to explore this direction.

**Comment by CDrummond** [comment]
@chrober Thanks. I note your enhancements all seem to use weighted similarity. Any chance the last.fm code could be adapted so it works with normal and forest? As its a secondary filtering, I see no reason why it could not. Likewise with using play counts - ideally that should be across all.

My thought being that the call to the mixer binary should just indicate the method to use (normal, forest, weighted) and then (optional) post processing with last.fm and/or play counts.

**Comment by chrober** [comment]
Yes, that makes sense to me. I'd be happy to contribute the Last.fm artist-similarity and play-count reranking so that both work with static weights, EIF, and adaptive weights. Would a contribution along those lines to lms-blissmixer be welcome?

**Comment by CDrummond** [comment]
Of course. Whilst play counts is not a feature of personally use, I can see it being useful to others. 

So, yeah, if you don't mind doing the work then that'd be great.

---

I'm currently working on a plugin to restore LMS item IDs after a wipe-and-scan. This should fix some SlimPing issues. After that I want to integrate Bliss with SlimPing to provide support for the open-subsonic similar tracks API.

**Comment by chrober** [comment]
Thanks! I have opened the PR here: https://github.com/CDrummond/lms-blissmixer/pull/20

I still haven’t had time to properly look into SlimPing yet, but it’s on my radar.

On my side I’m trying to get started with some `bliss-rs` analysis enhancements. I’ve started collecting/designing that work (https://chrober.github.io/bliss-similarity-design/) and already reached out to polochon-street (https://github.com/chrober/bliss-similarity-design/issues/1).

`Better Call Bliss` and `Bliss Mixer Lab` will likely serve as a bit of a playground for trying those ideas in practice before anything is proposed upstream.
